test/fail_compilation/test17892.d

```diff
@@ -16,13 +17,16 @@ struct S
     int[1] y;
     auto pointer1() return { return &x; }
     auto pointer2() return { return &y[0]; }
+    auto pointer3() return { return &y[0..1][0]; }
 }
 
 @safe int* testPointer(bool b)
 {
     S s;
     if (b)
         return s.pointer1();
-    else
+    else if (!b)
         return s.pointer2();
+    else
+        return s.pointer3();
```

Comment:
Use a switch if you'll be adding more cases? This branch should really error that its unreachable.

Reply:
Doesn't matter if it is unreachable or not. The purpose of the test is to get it to fail to compile with the correct message. Any change that changes that will not pass the autotester.